ixmp4/data/abstract/checkpoint.py

```diff
@@ -0,0 +1,140 @@
+from typing import Protocol
+
+import pandas as pd
+
+# TODO Import this from typing when dropping Python 3.11
+from typing_extensions import Unpack
+
+from ixmp4.data import types
+from ixmp4.data.abstract import annotations
+
+from . import base
+
+
+class EnumerateKwargs(
+    annotations.HasRunIdFilter,
+    annotations.HasTransactionIdFilter,
+    total=False,
+):
+    pass
+
+
+class Checkpoint(base.BaseModel, Protocol):
+    """ """
+
+    run__id: types.Integer
+    "Id of associated run."
+
+    transaction__id: types.Integer
+    "Id of associated transaction."
+
+    message: types.String
+    "Checkpoint message."
+
+    def __str__(self) -> str:
+        return f"<Checkpoint {self.id} message={self.message}>"
+
+
+class CheckpointRepository(
+    base.Creator,
+    base.Deleter,
+    base.Enumerator,
+    Protocol,
+):
+    def create(self, run__id: int, message: str) -> Checkpoint:
+        """Creates a checkpoint.
```

Comment:
```suggestion
        """Creates a Checkpoint.
```

To distinguish from the regular word.

Reply:
I'd do the same for 'run', 'model', and 'checkpoint' below.

Reply:
We dont capitalize these in all other parts of the codebase (at least the parts I wrote). I would argue that it makes sense to capitalize if we actually use a link to the class like `:class:\`pandas.Series\``, but in that case its probably better to change all occurances of this, not just the ones added in this PR